plugins/kubernetes-backend/src/service/KubernetesFanOutHandler.ts

```diff
@@ -345,11 +349,17 @@ export class KubernetesFanOutHandler {
         .filter(isString),
     );
 
-    const podMetrics = Array.from(namespaces).map(ns =>
-      this.fetcher.fetchPodMetricsByNamespace(clusterDetails, ns),
+    if (namespaces.size === 0) {
+      return [result, []];
+    }
+
+    const podMetrics = await this.fetcher.fetchPodMetricsByNamespaces(
+      clusterDetails,
+      namespaces,
     );
 
-    return Promise.all([result, Promise.all(podMetrics)]);
+    result.errors.concat(podMetrics.errors);
```

Comment:
OK I figured out the problem. From [these docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat),

> The `concat()` method is used to merge two or more arrays. This method does not change the existing arrays, but instead returns a new array. 

so it would make more sense to have

```suggestion
    result.errors = result.errors.concat(podMetrics.errors);
```

or similar, whatever style feels right. I'd say an error case in the unit tests KubernetesFetcher is required as well.

Reply:
Oh what a rookie mistake! 😂 Thanks for spotting this @jamieklassen. 

>  I'd say an error case in the unit tests KubernetesFetcher is required as well.

Totally, the truth is that we don't even have test cases for fetching pod metrics 😂. My intention was to have a POC and see if the community is ok with where I'm heading to. Since both @jhaals and you seem to be in favor of the change, I will definitely add more tests and do a few e2e tests on my local to make sure it works as intended.